app/src/main/java/com/github/se/project/ui/lesson/ConfirmedLesson.kt

```diff
@@ -231,27 +230,22 @@ private fun LessonDetailsCard(
 
 @Composable
 private fun MessageButton(
-    otherProfile: Profile,
     lesson: Lesson,
     isStudent: Boolean,
-    isConnected: Boolean
+    chatViewModel: ChatViewModel,
+    navigationActions: NavigationActions
 ) {
-  val context = LocalContext.current
+
   LessonActionButton(
       text = "Message ${if (isStudent) "Tutor" else "Student"}",
       onClick = {
-        if (isConnected) {
-          val intent =
-              Intent(Intent.ACTION_VIEW).apply {
-                data = Uri.parse("sms:${otherProfile.phoneNumber}")
-                putExtra("sms_body", "Hello, about our lesson ${formatDate(lesson.timeSlot)}...")
-              }
-          context.startActivity(intent)
-        } else {
-          Toast.makeText(
-                  context, context.getString(R.string.inform_user_offline), Toast.LENGTH_SHORT)
-              .show()
-        }
+        chatViewModel.createOrGetChannel(
+            lesson,
+            { channel ->
+              chatViewModel.setCurrentChannelId(channel.cid)
+              navigationActions.navigateTo(Screen.CHAT)
+            },
+            lesson.title)
```

Comment:
This new version is much more readable. Why was the isConnected check removed though?